src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -348,6 +349,303 @@ private Boolean decorationStep(final Structure.StructureBlock structureBlock)
+            if (job instanceof JobBuilder && ((JobBuilder) job).getStructure().getBlockInfo().tileentityData != null)
+            {
+                final TileEntityFlowerPot tileentityflowerpot = (TileEntityFlowerPot) world.getTileEntity(pos);
+                tileentityflowerpot.readFromNBT(((JobBuilder) job).getStructure().getBlockInfo().tileentityData);
+                world.setTileEntity(pos, tileentityflowerpot);
+            }
+        }
+        else
+        {
+            if (!world.setBlockState(pos, blockState, 0x03))
+            {
+                return false;
+            }
+        }
+
+        if (block instanceof BlockChest && job instanceof JobBuilder)
+        {
+            final BlockPos buildingLocation = ((JobBuilder) job).getWorkOrder().getBuildingLocation();
+            final AbstractBuilding building = this.getOwnBuilding().getColony().getBuilding(buildingLocation);
+            building.addContainerPosition(pos);
+        }
+
+        //It will crash at blocks like water which is actually free, we don't have to decrease the stacks we have.
+        if (isBlockFree(block, block.getMetaFromState(blockState)))
+        {
+            return true;
+        }
+
+        @Nullable final ItemStack stack = BlockUtils.getItemStackFromBlockState(blockState);
+        if (stack == null || stack == ItemStack.EMPTY)
+        {
+            Log.getLogger().error("Block causes NPE: " + blockState.getBlock());
+            return false;
+        }
+
+        final List<ItemStack> itemList = new ArrayList<>();
+        itemList.add(stack);
+        if (job instanceof JobBuilder && ((JobBuilder) job).getStructure() != null
+              && ((JobBuilder) job).getStructure().getBlockInfo() != null && ((JobBuilder) job).getStructure().getBlockInfo().tileentityData != null)
+        {
+            itemList.addAll(getItemStacksOfTileEntity(((JobBuilder) job).getStructure().getBlockInfo().tileentityData));
+        }
+
+        for (final ItemStack tempStack : itemList)
+        {
+            if (tempStack != null && tempStack != ItemStack.EMPTY)
+            {
+                final int slot = worker.findFirstSlotInInventoryWith(tempStack.getItem(), tempStack.getItemDamage());
+                if (slot != -1)
+                {
+                    new InvWrapper(getInventory()).extractItem(slot, 1, false);
+                    reduceNeededResources(tempStack);
+                }
+            }
+        }
+
+        if (Configurations.builderBuildBlockDelay > 0 && block != Blocks.AIR)
+        {
+            setDelay(Configurations.builderBuildBlockDelay);
+        }
+
+        return true;
+    }
+
+    /**
+     * Defines blocks that can be built for free.
+     *
+     * @param block    The block to check if it is free.
+     * @param metadata The metadata of the block.
+     * @return true or false.
+     */
+    public static boolean isBlockFree(@Nullable final Block block, final int metadata)
+    {
+        return block == null
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Reduce the number of conditional operators (7) used in the expression (maximum allowed 5). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1067)